### ISSUE TITLE
Empty query result using as_df ignores out_fields and return_geometry parameters.

### ISSUE THREAD

**Issue by jdcarls2** [Esri/arcgis-python-api#921]
**Describe the bug**
When a feature layer query returns no features, the resulting dataframe retains the full schema of the layer, even if the `out_fields` parameter is defined or `return_geometry` is set to `FALSE`. 

**To Reproduce**
```python
some_service.layers[0].query(where='1=0', out_fields=['field_one', 'field_two'], return_geometry=False, as_df=True)
```
Returns:
|objectid|field_one|field_two|field_three|all_the_rest|SHAPE|
|-|-|-|-|-|-|

Whereas if the query returns features, the parameters are honored.

```python
some_service.layers[0].query(out_fields=['field_one', 'field_two'], return_geometry=False, as_df=True)
```
Returns:
| |objectid|field_one|field_two|
|-|-|-|-|
|0|1|a_value|another_value|
|1|3|yet_another|None|

Alternately, doing the query first and using `.sdf` on the featureset *does* respect the fields.
```python
some_service.layers[0].query(where='1=0', out_fields=['field_one', 'field_two'], return_geometry=False).sdf
```
Returns:
|objectid|field_one|field_two|SHAPE|
|-|-|-|-|

**Expected behavior**
I would expect the fields listed in `out_fields` to be the only columns returned in my dataframe (besides `objectid` and, if I've asked for it, `SHAPE`), regardless of whether the query returns any features. There are cases when I do not know if a query will return features or not, and using the output dataframe as an input for other functions (especially joining and merging dataframes) can be problematic due to the potential for overlapping column names.

**Platform (please complete the following information):**
 - OS: Windows 10
 - ArcGIS Pro 2.7
 - Python API Version `1.8.3`

**Comment by mohi9282** [comment]
@jdcarls2 - thank you for reporting the issue. We will track it on geosaurus.

**Comment by mohi9282** [comment]
A fix for this issue was implemented with release v1.9.0.
